crates/bevy_ecs/src/core/archetype.rs

```diff
@@ -555,46 +553,4 @@ fn align(x: usize, alignment: usize) -> usize {
     (x + alignment - 1) & (!alignment + 1)
 }
 
-/// A hasher optimized for hashing a single TypeId.
-///
-/// TypeId is already thoroughly hashed, so there's no reason to hash it again.
-/// Just leave the bits unchanged.
-#[derive(Default)]
-pub(crate) struct TypeIdHasher {
-    hash: u64,
-}
-
-impl Hasher for TypeIdHasher {
-    fn write_u64(&mut self, n: u64) {
-        // Only a single value can be hashed, so the old hash should be zero.
-        debug_assert_eq!(self.hash, 0);
-        self.hash = n;
-    }
-
-    // Tolerate TypeId being either u64 or u128.
-    fn write_u128(&mut self, n: u128) {
-        debug_assert_eq!(self.hash, 0);
-        self.hash = n as u64;
-    }
-
-    fn write(&mut self, bytes: &[u8]) {
-        debug_assert_eq!(self.hash, 0);
-
-        // This will only be called if TypeId is neither u64 nor u128, which is not anticipated.
-        // In that case we'll just fall back to using a different hash implementation.
-        let mut hasher = AHasher::default();
-        hasher.write(bytes);
-        self.hash = hasher.finish();
-    }
-
-    fn finish(&self) -> u64 {
-        self.hash
-    }
-}
-
-/// A HashMap with TypeId keys
-///
-/// Because TypeId is already a fully-hashed u64 (including data in the high seven bits,
-/// which hashbrown needs), there is no need to hash it again. Instead, this uses the much
-/// faster no-op hash.
-pub(crate) type TypeIdMap<V> = HashMap<TypeId, V, BuildHasherDefault<TypeIdHasher>>;
+pub(crate) type TypeIdMap<V> = HashMap<TypeId, V, fxhash::FxBuildHasher>;
```

Comment:
Can we leave a comment here mentioning why we aren't just using `RandomState`, and perhaps mentioning `TypeIdHasher`.

Reply:
>perhaps mentioning TypeIdHasher
I don't think that there is a reason to mention deleted code in comments. It would just distract people unneccessary.